sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -98,6 +102,8 @@ typedef struct _nib_onl_entry {
      * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ARSM != 0.
      */
     uint8_t l2addr[GNRC_IPV6_NIB_L2ADDR_MAX_LEN];
+    evtimer_msg_event_t reach_timeout;  /**< Event for @ref GNRC_IPV6_NIB_REACH_TIMEOUT */
+    evtimer_msg_event_t delay_timeout;  /**< Event for @ref GNRC_IPV6_NIB_DELAY_TIMEOUT */
```

Comment:
Okay, thanks for this informative description. I got a better picture of the proposed events now and I am glad you merged some of them (ucast/mcast NS/NA, e.g.). I guess we have to see how this develops. Just keep in mind that for the 6LR case, there needs to be enough room for an actual routing protocol on top.

Speaking of lower tier power-houses: ND+RPL should fit (with reasonable data structure sizes) into `iotlab-m3` nodes (64kB RAM). According to your back-of-the-envelope calculation above, ND's RAM usage seems to be moderate and should not pose an issue for these power-houses.